### ISSUE TITLE
[RFC]: proposal: bump Jenkins 'inactivity' timeout from 5mins to 30mins

### ISSUE THREAD

**Issue by grahamwhaley** [kata-containers/ci#97]
Our current Jenkins setup has an 'inactivity timer' (meaning, if there is no console output from the build job, the job is killed off as hung up and failed), set to 5 minutes (300s) at present.
Having the timeout is a good thing, but having it at a relatively short period has caused me aggravation on a couple of 'chronic' PRs to try and reduce out CI log sizes:
https://github.com/kata-containers/tests/pull/994
https://github.com/kata-containers/tests/pull/810

In both those PRs we tried to work around the inactivity timer kicking in on 'long running and noisy tasks' (such as building the linux kernel from sources), by adding a background 'heatbeat ticker' with some shell magic. This caused us a couple of problems though, making both of those PRs not tenable right now.

I propose we just bump the inactivity timer to a suitably large time (say 30 minutes, 1800s), and thus remove the heart beat ticker code from those PRs. That will enable us to land one of the PRs, and simplify the rework and general code on the other I believe.

My general premise behind being happy with bumping the timeout to a large value is:
- most of the time (95+%), the timer will never kick in, as the CI jobs will pass
- if a CI job is going to fail due to a hang up, finding out in either 5 or 30 minutes imho will make little difference to our workflow. If it is a PR specific hang up, it will need fixing anyway, which will take time. If it is a global issue (say another PR has broken the CI infra, as we have seen happen before), then finding out the whole CI system is broken in 5 or 30 minutes I feel is of little cost for the rare case.

If we are not happy bumping the timeout (and I can believe we may not be, but would like to hear some of the justfications etc.), then I'll just close the two pending PRs for the moment as being 'too hard to fix', and leave the Issues open for them so we have some history/trail in place.

/cc @chavafg @sboeuf @egernst @gnawux @nitkon @Pennyzct for thoughts.
also @jodh-intel @marcov who've been staring at the PRs in question.

**Comment by grahamwhaley** [comment]
This timeout is just for 'console inactivity'. So, it's not the 'total build time', it is purely any 'single consecutive time when nothing came out the console' - as long as we set it high enough to cover any `chronic`d phase of the CI not generating any output, on the slowest build node, then we should be good.

For a view of how long builds are actually taking you can:

- look in the RHS column of the build jobs on the [main jenkins master page](http://jenkins.katacontainers.io/)  and see the 'Last duration', which gives an indication of how long builds are taking (if you squint to remove any failures etc.) - for instance, right now they are hanging around the 1h20m time.
- for each build job type, you can click on the 'trend' link (and, that is new to me...), which will give you a list of the historic build status and times, and a pretty little graph as well like [this](http://jenkins.katacontainers.io/job/kata-containers-runtime-ubuntu-16-04-PR/buildTimeTrend).
- For individual build nodes (which works for the metrics bare metal, but not really for our cloud based build nodes), you can view the build history of the node like [this for the metrics slave](http://jenkins.katacontainers.io/computer/x86_packet01/builds)

I think that 'trend' graph is going to be the most useful (my new Jenkins find of the week @jodh-intel @chavafg ;-) )

**Comment by jodh-intel** [comment]
Oh - sorry, misread! Yeah, this seems fine with me then. Ta for pointers.

**Comment by chavafg** [comment]
the only downside I see here is that we would waste 30 minutes of a resource when this happens, but as @grahamwhaley mentions, it is a rare case when we hit a timeout. 

@grahamwhaley do you think 30 mins is the most suitable value for the timeout? or can it be less than that? e.g. 10 min?

**Comment by grahamwhaley** [comment]
Hi @chavafg 
The min time we could set would be the longest item in the build that we'd like to silence with a `chronic`. I suspect that is going to be building the kernel from source. The easiest way to get a handle on that is probably to go find a CI build log and look at the timestamps, and then maybe add 50% on to that time to cater for variance in build slaves and other glitches (network speed etc.). Let me see if I can find one....

Oh, we used to have timestamps in the log files in Jenkins, but I didn't see them in the 16.04 build log I just peeked at - maybe we dropped them at some point in our upgrade/transitions... indeed, the 'add timestamps to logs' is disabled in the build jobs. If we want to find a time then maybe we can enable that on one build job, do a build (rebuild), capture the logs, and disable again @chavafg ?
